internal/cli/atlas_alerts_acknowledge.go

```diff
@@ -52,20 +53,18 @@ func (opts *atlasAlertsAcknowledgeOpts) Run() error {
 }
 
 func (opts *atlasAlertsAcknowledgeOpts) newAcknowledgeRequest() *atlas.AcknowledgeRequest {
-	until := opts.until
-
 	// To acknowledge an alert “forever”, set the field value to 100 years in the future.
-	if until == "" {
-		until = time.Now().AddDate(100, 1, 1).Format(time.RFC3339)
+	if opts.forever {
+		opts.until = time.Now().AddDate(100, 1, 1).Format(time.RFC3339)
 	}
 
 	return &atlas.AcknowledgeRequest{
-		AcknowledgedUntil:      until,
+		AcknowledgedUntil:      opts.until,
```

Comment:
[q] do we need to update the client? this field has an omit empty so I'm wondering how is it working with blank?

Reply:
I don't think we have to update the atlas client. The field is set to `nil` when the `until` flag is black which is what we want here.

Reply:
`AcknowledgedUntil` is not a pointer so it can't be nil, only empty ("")
when we serialize it it would only be something like an empty object `{}` so maybe the API is assuming that since there's no key is null?

Reply:
I thought that the serialization of an empty string with `omitempty`  was `nil` since I got `null` when I printed the value of until in the server.

Reply:
`omitempty` means that it won't serialize/deserialize that key not about its content, the content defines the rule for `omitempty`, which in the case of string is the empty string ""

Reply:
Yes, I noticed it, I was reading this article https://www.sohamkamani.com/golang/2018-07-19-golang-omitempty/ that explains it very well. Regarding your question, as you previously said, the only explanation seems to be that the API set the object to null when there is no key (which makes sense).

Reply:
yes, omitempty is confusing, I know, is one of the things took me the most to get on go, and we can trip a lot with it
if you are sure this works as expected we can merge but we need to be aware that a change on the API can break this command

Reply:
I am quite confident that it is working as expected but we can make `AcknowledgedUntil` in the atlas client a pointer just to be sure that this behaviour is not going to change in the future. Let me know if you want me to open a PR in the altlas go client.

Reply:
@andreaangiolillo sorry, was busy and forgot about this, up to you, I'm ok to merge this as it is and we can improve the client later